### PR TITLE
Added option to disable deprecation warnings

### PR DESCRIPTION
**Pull request by jankcat**
## Proposed changes

I added a config option to enable/disable deprecation warnings. The config option is `deprecationWarnings`, and is set to `true` by default. This means that by default, deprecation warnings will be displayed. Setting `deprecationWarnings` to `false` will cause deprecation warnings to not be displayed. Just like the simulations.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

To recap... `deprecationWarnings` option...

### CODE REVIEW

File: examples/wdio/wdio.mocha.conf.js [hunk 1]
```diff
@@ -56,5 +57,5 @@ exports.config = {
     onComplete: function() {
         console.log('that\'s it');
     }
-
-};
\ No newline at end of file
+
```
Comment (christian-bromann) [inline examples/wdio/wdio.mocha.conf.js]: I guess this is not needed 😉

File: docs/guide/testrunner/gettingstarted.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ Options:
   --watch               watch specs for changes
   --logLevel, -l        level of logging verbosity (default: silent)
   --coloredLogs, -c     if true enables colors for log output (default: true)
+  --deprecationWarnings if true use of deprecated commands will output a warning (default: true)
```
Comment (christian-bromann) [inline docs/guide/testrunner/gettingstarted.md]: if you want to add it here you need to add support for it in `cli.js`. If you ask me .. not needed

File: lib/helpers/deprecationWarning.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export default function deprecate (commandName) {
+    if (!global.browser.options.deprecationWarnings) return
```
Comment (christian-bromann) [inline lib/helpers/deprecationWarning.js]: This wouldn't work in standalone mode. Let's call all `deprecate` functions with `this` context and access that option in that way.

Comment (christian-bromann) [inline lib/helpers/deprecationWarning.js]: or apply the option as argument

File: docs/guide/testrunner/gettingstarted.md (hunk 2 above)
Comment (jankcat) [inline docs/guide/testrunner/gettingstarted.md]: Good call. Will fix.

File: lib/helpers/deprecationWarning.js (hunk 3 above)
Comment (jankcat) [inline lib/helpers/deprecationWarning.js]: Oh shoot. Thanks for the tip. Will make it an argument then.

File: examples/wdio/wdio.mocha.conf.js (hunk 1 above)
Comment (jankcat) [inline examples/wdio/wdio.mocha.conf.js]: Haha WebStorm automatically makes sure files end in a newline... I can take the line out after-the-fact on github if you want

Comment (jankcat) [inline examples/wdio/wdio.mocha.conf.js]: Oh wait, `insert_final_newline = true` is part of the `.editorconfig`, so this is intentional I suppose.

**Comment by jankcat** [comment]
@christian-bromann requested changes made

**Comment by jankcat** [comment]
To show it works in standalone:
The [moveTo.js](https://github.com/webdriverio/webdriverio/blob/master/test/spec/unit/moveTo.js) unit test uses a deprecated function in standalone mode.
screen shot 2017-05-09 at 11 20 44 am

When I update this line:
```
const client = remote({})
```
To this:
```
const client = remote({deprecationWarnings: false})
```
The UT output looks as expected:
screen shot 2017-05-09 at 11 19 02 am

**Comment by webOS101** [comment]
Just food for thought -- would it make sense to pass an array of the methods to suppress instead, so that new deprecation warnings don't get stifled unexpectedly?

File: examples/wdio/wdio.mocha.conf.js (hunk 1 above)
Comment (christian-bromann) [inline examples/wdio/wdio.mocha.conf.js]: I was talking about the unnecessary line (59). Let's remove this

File: lib/protocol/context.js [hunk 4]
```diff
@@ -25,6 +25,6 @@ export default function context (id) {
         data.name = id
     }
 
-    deprecate('context')
+    deprecate('context', this.options)
```
Comment (christian-bromann) [inline lib/protocol/context.js]: Can we remove the deprecate warning here? This is an Appium command and won't be removed anytime soon.

File: lib/protocol/contexts.js [hunk 5]
```diff
@@ -12,6 +12,6 @@
 import deprecate from '../helpers/deprecationWarning'
 
 export default function contexts () {
-    deprecate('contexts')
+    deprecate('contexts', this.options)
```
Comment (christian-bromann) [inline lib/protocol/contexts.js]: same here

**Comment by jankcat** [comment]
@christian-bromann I have made the requested changes.
